### PR TITLE
Accounts Aging reports UI and API integration

### PR DESCRIPTION
**Pull request by Shruti-Apte**
## Notion card
https://www.notion.so/saeloun/Create-the-frontend-designs-for-Accounts-Aging-report-09c39ab08a7244edbb5503aad9a30c10
## Summary

- Accounts aging report section added.
- Table with alphabetical order, Filter with search, table row with total and summary dashboard  added.
## Preview

Screenshot 2022-12-20 at 1 56 09 PM

Screenshot 2022-12-20 at 1 55 04 PM

Screenshot 2022-12-20 at 1 55 16 PM

https://www.loom.com/share/967df4701ddf4a518e0458b47ce4bc4b
## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)

## How Has This Been Tested?

- Checked Filtering of the reports. 
- Checked empty states.
- Pagination is not implemented.

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have annotated changes in the PR that are relevant to the reviewer

### CODE REVIEW

**Review by apoorv1316 — CHANGES_REQUESTED** [review]
@Shruti-Apte Please see the loom video https://www.loom.com/share/cdaa331d7cc443bc958ad5e3c8da1901?focus_title=1&muted=1&from_recorder=1

File: app/javascript/src/components/Reports/accountsAging/Filters/index.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    }
+  };
+
+  const handleApply = () => {
+    setSelectedFilter(selectedClients);
+    setFilterCounter(selectedClients.length);
+    setIsFilterVisible(false);
+  }; //eslint-disable-line
+
+  const handleReset = () => {}; //eslint-disable-line
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Reports/accountsAging/Filters/index.tsx:61]: @Shruti-Apte Please remove `handleReset` if it is not needed

File: app/javascript/src/components/Reports/accountsAging/Filters/index.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
+import React, { useState, useEffect } from "react";
+
+import { useDebounce } from "helpers";
+import { XIcon, FilterIcon, PlusIcon, MinusIcon, SearchIcon } from "miruIcons";
+import { SidePanel, Button } from "StyledComponents";
+
+import CustomCheckbox from "common/CustomCheckbox";
+import { useEntry } from "components/Reports/context/EntryContext";
+
+const FilterSideBar = ({
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Reports/accountsAging/Filters/index.tsx:10]: @Shruti-Apte Component name and folder name should be the same if the file is `index.tsx` for better-naming conventions
eg: 
`app/javascript/src/components/Reports/accountsAging/Filters/index.tsx`

then component name `Filters`

File: app/javascript/src/components/Reports/accountsAging/index.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from "react";
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Reports/accountsAging/index.tsx:1]: @Shruti-Apte Here folder name should be `AccountsAging`

File: app/javascript/src/components/Reports/api/accountsAging.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import accountsAgingApi from "apis/reports/accountsAging";
+import { Toastr } from "common/Toastr";
+
+const getReportData = async ({ setClientList, setTotalAmount }) => {
+  try {
+    const res = await accountsAgingApi.get();
+    setClientList(res.data.report.clients);
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Reports/api/accountsAging.ts:8]: @Shruti-Apte Destructure the response, assign it to variables and use them.
Remove the unnecessary commented code.

File: app/javascript/src/components/Reports/reportList/index.tsx [hunk 5]
```diff
@@ -39,16 +40,23 @@ const listDetails = [
...
+  {
+    icon: accounts_aging,
+    title: "Accounts Aging",
+    description: "Find out which client have been taking a long time to pay",
+    url: "accounts-aging",
+    show: true,
+  },
 ];
 
 const ReportsList = () => (
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Reports/reportList/index.tsx:52]: @Shruti-Apte Here the folder name should be List

File: app/javascript/src/components/Reports/outstandingInvoices/index.tsx [hunk 6]
```diff
@@ -8,6 +8,7 @@ import Filters from "./Filters";
 import { OutstandingOverdueInvoice } from "./interface";
 
 import getReportData from "../api/outstandingOverdueInvoice";
+import AccountsAgingReportContext from "../context/AccountsAgingReportContext";
 import EntryContext from "../context/EntryContext";
 import RevenueByClientReportContext from "../context/RevenueByClientContext";
 import TimeEntryReportContext from "../context/TimeEntryReportContext";
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Reports/outstandingInvoices/index.tsx:14]: @Shruti-Apte Here the folder name and component should be same.

File: app/javascript/src/components/Reports/revenueByClient/index.tsx [hunk 7]
```diff
@@ -8,6 +8,7 @@ import Filters from "./Filters";
 import { RevenueByClients } from "./interface";
 
 import getReportData from "../api/revenueByClient";
+import AccountsAgingReportContext from "../context/AccountsAgingReportContext";
 import EntryContext from "../context/EntryContext";
 import OutstandingOverdueInvoiceContext from "../context/outstandingOverdueInvoiceContext";
 import TimeEntryReportContext from "../context/TimeEntryReportContext";
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Reports/revenueByClient/index.tsx:14]: @Shruti-Apte Here folder name and component should be same for `RevenueByClientReport`

File: app/javascript/src/components/Reports/timeEntry/index.tsx [hunk 8]
```diff
@@ -5,6 +5,7 @@ import { sendGAPageView } from "utils/googleAnalytics";
 
 import { applyFilter, getQueryParams } from "../api/applyFilter";
 import Container from "../Container";
+import AccountsAgingReportContext from "../context/AccountsAgingReportContext";
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Reports/timeEntry/index.tsx:8]: @Shruti-Apte Here Folder name and component should be the same `TimeEntryReports` or make it `TimeEntry` for the folder name and keep component`TimeEntryReports`

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Comment by apoorv1316** [comment]
@Shruti-Apte App is breaking. Please fix
Screenshot 2023-01-02 at 12 46 36 PM

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Comment by supriya3105** [comment]
@Shruti-Apte Lets close the review comments from Apoorv and Prasanth on this PR today.

File: app/javascript/src/StyledComponents/SummaryDashboard.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+type SummaryDashboardProps = {
+  summaryList: any;
+  currency: any;
+};
+
+const DEFAULT_STYLE =
+  "mt-6 px-10 py-10 flex flex-col lg:flex-row lg:overflow-x-auto rounded-2xl bg-miru-han-purple-1000  text-white";
+
+const formattedAmount = (amount, baseCurrency) =>
```
[the remark this answers is not in the document]
Comment (Shruti-Apte) [reply app/javascript/src/StyledComponents/SummaryDashboard.tsx:14]: can we create a different refactor ticket for this ?? currencyFormat function is used in so many files @prasanthchaduvula @supriya3105

[the remark this answers is not in the document]
Comment (supriya3105) [reply app/javascript/src/StyledComponents/SummaryDashboard.tsx:14]: @prasanthchaduvula I have created a new ticket for this. Please approve this PR is all other review comments have been addressed.

**Review by prasanthchaduvula — APPROVED** [review]

**Review by apoorv1316 — CHANGES_REQUESTED** [review]
@Shruti-Apte Something is wrong with the aging reports. Not able to see clients and their data.
Screenshot 2023-01-05 at 3 02 16 PM

**Comment by Shruti-Apte** [comment]
> @Shruti-Apte Something is wrong with the aging reports. Not able to see clients and their data. Screenshot 2023-01-05 at 3 02 16 PM

Fixed!

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**